### DISCUSSION TITLE
Restart React-Joyride

### DISCUSSION THREAD

**Discussion by venumanthena** [gilbarbara/react-joyride#646] [Q&A] [answered]
I have a button on clicking  which I am showing the tour, but it only plays once. Is there a way to restart the react joyride tour on clicking the button again. The methods mentioned in the docs this.joyride.reset() etc are not working. Once the tour is over, I am not able to start it again. It only plays once.

**Comment by venumanthena** [comment, marked as answer]
Got fixed thanks

**Reply by jiyang-sas** [reply]
I am having the same problem now. How did you fix it?

**Reply by venumanthena** [reply]
On clicking of your button just make the stepIndex to 0 and run true. This should solve your issue.

**Reply by jiyang-sas** [reply]
em, I did that but it did not work.
